app/src/sharedTest/java/org/oppia/android/app/topic/conceptcard/ConceptCardFragmentTest.kt

```diff
@@ -363,6 +370,112 @@ class ConceptCardFragmentTest {
     }
   }
 
+  @Test
+  fun testConceptCardFragment_launchSeveralConceptCardsWithSameSkill_onlyTheFirstOne() {
+    launchTestActivity().use { scenario ->
+      scenario.onActivity { activity ->
+        val fragment = ConceptCardFragment.bringToFrontOrCreateIfNew(
+          TEST_SKILL_ID_0,
+          ProfileId.getDefaultInstance(),
+          activity.supportFragmentManager
+        )
+        Assert.assertNotNull(fragment)
+        Assert.assertNull(
+          ConceptCardFragment.bringToFrontOrCreateIfNew(
+            TEST_SKILL_ID_0,
+            ProfileId.getDefaultInstance(),
+            activity.supportFragmentManager
+          )
+        )
+        Assert.assertEquals(1, activity.supportFragmentManager.fragments.size)
+        Assert.assertEquals(fragment, activity.supportFragmentManager.fragments[0])
+      }
+    }
+  }
+
+  @Test
+  fun testConceptCardFragment_twoConceptCards_secondOneReplacesFirstOne() {
+    launchTestActivity().use { scenario ->
+      scenario.onActivity { activity ->
+        val fragmentSkill0 = ConceptCardFragment.bringToFrontOrCreateIfNew(
+          TEST_SKILL_ID_0,
+          ProfileId.getDefaultInstance(),
+          activity.supportFragmentManager
+        )
+        Assert.assertNotNull(fragmentSkill0)
+        val fragmentSkill1 =
+          ConceptCardFragment.bringToFrontOrCreateIfNew(
+            TEST_SKILL_ID_1,
+            ProfileId.getDefaultInstance(),
+            activity.supportFragmentManager
+          )
+        Assert.assertNotNull(fragmentSkill1)
+        Assert.assertEquals(1, activity.supportFragmentManager.fragments.size)
+        Assert.assertEquals(fragmentSkill1, activity.supportFragmentManager.fragments[0])
+      }
+    }
+  }
+
+  @Test
+  fun testConceptCardFragment_mixedWithNonConceptCardFragments_onlyConceptCardsAreAffected() {
+    launchTestActivity().use { scenario ->
+      scenario.onActivity { activity ->
+        val fragmentSkill0 = ConceptCardFragment.bringToFrontOrCreateIfNew(
+          TEST_SKILL_ID_0,
+          ProfileId.getDefaultInstance(),
+          activity.supportFragmentManager
+        )
+        Assert.assertNotNull(fragmentSkill0)
+        val fragmentSkill1 =
+          ConceptCardFragment.bringToFrontOrCreateIfNew(
+            TEST_SKILL_ID_1,
+            ProfileId.getDefaultInstance(),
+            activity.supportFragmentManager
+          )
+        Assert.assertNotNull(fragmentSkill1)
+        fragmentSkill0!!.showNow(activity.supportFragmentManager, fragmentSkill0.tag)
+        Assert.assertEquals(2, activity.supportFragmentManager.fragments.size)
+        ConceptCardFragment.dismissAll(activity.supportFragmentManager)
+        Assert.assertEquals(0, activity.supportFragmentManager.fragments.size)
+      }
+    }
+  }
+
+  @Test
+  fun testConceptCardFragment_severalConceptCards_dismissAll() {
```

Comment:
Perhaps add a test for dismissAll when there are no cards?

Reply:
Done